modules/vectra.py

```diff
@@ -54,22 +56,35 @@ def request_handler(self, *args, **kwargs):
                 raise HTTPException(response)
         return request_handler
 
-def validate_api_v2(func):
-    def api_validator(self, *args, **kwargs):
-        if self.version >= 2:
+def renew_access_token(func):
+    def wrapper(self, *args, **kwargs):
+        try:
             return func(self, *args, **kwargs)
-        else:
-            raise NotImplementedError('Method only accessible via v2 of API')
-
-    return api_validator
-
-def deprecation(message):
-    warnings.warn(message, PendingDeprecationWarning)
-
-def param_deprecation(key):
-    message = f'{key} will be deprecated with Vectra API v1 which will be annouced in an upcoming release'
-    warnings.warn(message, PendingDeprecationWarning)
-
+        except HTTPUnauthorizedException as e:
+            # To calculate token expiration we take 10s margin
+            if not self.refresh_token or (self.access_token_validity-time() > 10.0 and self.refresh_token_validity-time()>10.0):
```

Comment:
I would check your formatting throughout this file.  Running black should help.  We have cases where we have the operator missing spaces to the right and/or left.

It may be good to run black against this file as a separate PR:
https://pypi.org/project/black/